### ISSUE TITLE
MySQL connector throws exception on Linux build

### ISSUE THREAD

**Issue by drakeee** [altmp/altv#538]
**Current behavior**
When I try to connect to my MySQL server (with MysqlConnector or with Mysql.Data) with the newest build (#1160) it throws the following exception:

With MysqlConnector (v0.64.2):
```
[18:36:12] alt:V Server, build #1160, branch release
[18:36:12] Starting alt:V Server on 0.0.0.0:7788
[18:36:13] coreclr-module: version found: 3.1.3
[18:36:13] coreclr-module: greatest version: 3.1.3
[18:36:13] Loading resource *
[18:36:13] [*] Starting gamemode
[18:36:13] [SQL] Initialization connection with SQL server.
[18:36:13] [SQL] Connection reference: MySql.Data.MySqlClient.MySqlConnection
[SQL] Can't connect to MySQL server: Exception of type 'System.OutOfMemoryException' was thrown.
[SQL]    at System.Threading.TimerQueue.CreateAppDomainTimer(UInt32 dueTime, Int32 id)
   at System.Threading.TimerQueue.EnsureTimerFiresBy(UInt32 requestedDuration)
   at System.Threading.TimerQueue.UpdateTimer(TimerQueueTimer timer, UInt32 dueTime, UInt32 period)
   at System.Threading.TimerQueueTimer.Change(UInt32 dueTime, UInt32 period)
   at System.Threading.CancellationTokenSource.InitializeWithTimer(Int32 millisecondsDelay)
   at System.Threading.CancellationTokenSource..ctor(TimeSpan delay)
   at MySql.Data.MySqlClient.MySqlConnection.CreateSessionAsync(ConnectionPool pool, Int32 startTickCount, Nullable`1 ioBehavior, CancellationToken cancellationToken) in /_/src/MySqlConnector/MySql.Data.MySqlClient/MySqlConnection.cs:line 643
   at MySql.Data.MySqlClient.MySqlConnection.OpenAsync(Nullable`1 ioBehavior, CancellationToken cancellationToken) in /_/src/MySqlConnector/MySql.Data.MySqlClient/MySqlConnection.cs:line 319
   at MySql.Data.MySqlClient.MySqlConnection.Open() in /_/src/MySqlConnector/MySql.Data.MySqlClient/MySqlConnection.cs:line 285
   at *.Systems.SQL.Connect()
[18:36:14] Loaded resource *
[18:36:14] Starting HTTP server on 0.0.0.0:7788
[18:36:14] Server started
```

With Mysql.Data (v8.0.20):
```
[19:51:56] alt:V Server, build #1160, branch release
[19:51:56] Starting alt:V Server on 0.0.0.0:7788
[19:51:56] coreclr-module: version found: 3.1.3
[19:51:56] coreclr-module: version found: 5.0.0-preview.3.20214.6
[19:51:56] coreclr-module: greatest version: 5.0.0-preview.3.20214.6
[19:51:56] Loading resource *
[19:51:56] [*] Starting gamemode
[19:51:56] [SQL] Initialization connection with SQL server.
[19:51:56] [SQL] Connection reference: MySql.Data.MySqlClient.MySqlConnection
[19:51:57] [SQL] TypeInitializationException: The type initializer for 'MySql.Data.MySqlClient.MySqlPoolManager' threw an exception.
[19:51:57] [SQL]    at MySql.Data.MySqlClient.MySqlConnection.Open()
   at *.Systems.SQL.Connect()
[19:51:57] Loaded resource *
[19:51:57] Starting HTTP server on 0.0.0.0:7788
[19:51:57] Server started
```
Strange enough that if i use build 1153 binary with build 1160 files everything works as expected.
Also this issue only related to Linux, Windows doesn't reproduce this exceptions.

OP and SQL specs:
```
mysql  Ver 15.1 Distrib 10.3.22-MariaDB, for debian-linux-gnu (x86_64) using readline 5.2

Linux s1 4.19.0-8-amd64 #1 SMP Debian 4.19.98-1 (2020-01-26) x86_64 GNU/Linux
Distributor ID: Debian
Description:    Debian GNU/Linux 10 (buster)
Release:        10
Codename:       buster
```

**Expected behavior**
It should connect to the MySQL server.

**Comment by KubasGC** [comment]
For me this error also happening, but with different ending - server crash.

```
[19:55:29] alt:V Server, build #1159, branch dev
[19:55:29] Starting [DEV] Testing server | Vibe Role Play on 0.0.0.0:7789
[19:55:29] coreclr-module: version found: 5.0.0-preview.3.20214.6
[19:55:29] coreclr-module: version found: 3.1.1
[19:55:29] coreclr-module: greatest version: 5.0.0-preview.3.20214.6
[19:55:29] Loading resource vrp
4/30/2020 7:55:29 PM | [INFO] Configuring Autofac & database...
30.04.2020 19:55:30 | [ERROR] System.TypeInitializationException: The type initializer for 'MySql.Data.MySqlClient.MySqlPoolManager' threw an exception.
 ---> System.OutOfMemoryException: Exception of type 'System.OutOfMemoryException' was thrown.
   at System.Threading.TimerQueue.CreateAppDomainTimer(UInt32 dueTime, Int32 id)
   at System.Threading.TimerQueue.EnsureTimerFiresBy(UInt32 requestedDuration)
   at System.Threading.TimerQueue.UpdateTimer(TimerQueueTimer timer, UInt32 dueTime, UInt32 period)
   at System.Threading.TimerQueueTimer.Change(UInt32 dueTime, UInt32 period)
   at System.Threading.TimerQueueTimer..ctor(TimerCallback timerCallback, Object state, UInt32 dueTime, UInt32 period, Boolean flowExecutionContext)
   at System.Threading.Timer.TimerSetup(TimerCallback callback, Object state, UInt32 dueTime, UInt32 period, Boolean flowExecutionContext)
   at System.Threading.Timer..ctor(TimerCallback callback, Object state, Int32 dueTime, Int32 period, Boolean flowExecutionContext)
   at System.Threading.Timer..ctor(TimerCallback callback, Object state, Int32 dueTime, Int32 period)
   at MySql.Data.MySqlClient.MySqlPoolManager..cctor()
   --- End of inner exception stack trace ---
   at MySql.Data.MySqlClient.MySqlPoolManager.GetPool(MySqlConnectionStringBuilder settings)
   at MySql.Data.MySqlClient.MySqlConnection.Open()
   at DbUp.Engine.Transactions.DatabaseConnectionManager.OperationStarting(IUpgradeLog upgradeLog, List`1 executedScripts)
   at DbUp.Engine.UpgradeEngine.GetScriptsToExecute()
   at VRP.DAL.Migrations.MigrationService.UpdateDatabase() in /opt/buildagent/work/499ea41f091e36c3/VRP.DAL.Migrations/MigrationService.cs:line 18
   at VRP.GM.VibeResource.ConfigureAutofacAndDatabase() in /opt/buildagent/work/499ea41f091e36c3/VRP.GM/VibeResource.cs:line 174
30.04.2020 19:55:30 | [ERROR] There is an error with MigrationsService
30.04.2020 19:55:30 | [INFO] Configuring Sentry...
30.04.2020 19:55:30 | [INFO] Loading networking entity...
[19:55:30] < ==== UNHANDLED EXCEPTION ==== >
 Received an unhandled exception from : System.OutOfMemoryException: Exception of type 'System.OutOfMemoryException' was thrown.
   at System.Net.HttpWebRequest.GetResponse()
   at VRP.GM.VibeResource.GetPublicIp() in /opt/buildagent/work/499ea41f091e36c3/VRP.GM/VibeResource.cs:line 278
   at VRP.GM.VibeResource.ConfigureNetworkingEntity() in /opt/buildagent/work/499ea41f091e36c3/VRP.GM/VibeResource.cs:line 137
   at VRP.GM.VibeResource.OnStart() in /opt/buildagent/work/499ea41f091e36c3/VRP.GM/VibeResource.cs:line 100
30.04.2020 19:55:30 | [ERROR] System.OutOfMemoryException: Exception of type 'System.OutOfMemoryException' was thrown.
   at System.Net.HttpWebRequest.GetResponse()
   at VRP.GM.VibeResource.GetPublicIp() in /opt/buildagent/work/499ea41f091e36c3/VRP.GM/VibeResource.cs:line 278
   at VRP.GM.VibeResource.ConfigureNetworkingEntity() in /opt/buildagent/work/499ea41f091e36c3/VRP.GM/VibeResource.cs:line 137
   at VRP.GM.VibeResource.OnStart() in /opt/buildagent/work/499ea41f091e36c3/VRP.GM/VibeResource.cs:line 100
Out of memory.
./start.sh: line 4: 28411 Aborted                 ./altv-server
```


So - Out Of Memory Exception is firing first after using MySql.Data (which is used by DbUp package)
Next, error is firing on method GetPublicIp():

```cs
private string GetPublicIp()
{
     var request = WebRequest.Create("http://checkip.dyndns.org/");
    var response = request.GetResponse();
    var stream = new StreamReader(response.GetResponseStream());
    var direction = stream.ReadToEnd();
    stream.Close();
    response.Close(); //Search for the ip in the html
    direction = StringHelper.StripHtmlTags(direction);
    var first = direction.IndexOf("Address: ", StringComparison.Ordinal) + 9;
    var last = direction.LastIndexOf("", StringComparison.Ordinal);
    direction = direction.Substring(first, last - first);
    return direction;
}
```

then server is crashing.


```
Distributor ID: Debian
Description:    Debian GNU/Linux 9 (stretch)
Release:        9
Codename:       stretch
```
```
Linux Vibe-Role-Play 4.9.0-8-amd64 #1 SMP Debian 4.9.144-3.1 (2019-02-19) x86_64 GNU/Linux
```

```
10.3.14-MariaDB-1:10.3.14+maria~stretch - mariadb.org binary distribution
```

For Windows build everything looks to be good.

**Comment by TheMysteriousVincent** [comment]
Same thing happening to me since the last update, but to the assembling of Autofac IoC containers.

```
Unhandled exception. Autofac.Core.DependencyResolutionException: An exception was thrown while activating Eisengrind.FFst.Server.Components.Banking.BankingComponent -> Eisengrind.FFst.Server.Core.Storage.BankAccounts.MongoDB.Database -> Eisengrind.FFst.Server.Core.Storage.BankOrders.MongoDB.Database.
---> Autofac.Core.DependencyResolutionException: An exception was thrown while invoking the constructor 'Void .ctor(Eisengrind.FFst.Server.Core.Database.MongoDB.IFFDatabase)' on type 'Database'.
---> System.OutOfMemoryException: Exception of type 'System.OutOfMemoryException' was thrown.
at System.Threading.TimerQueue.CreateAppDomainTimer(UInt32 dueTime, Int32 id)
at System.Threading.TimerQueue.EnsureTimerFiresBy(UInt32 requestedDuration)
at System.Threading.TimerQueueTimer.Change(UInt32 dueTime, UInt32 period)
at System.Threading.Timer.Change(Int64 dueTime, Int64 period)
at System.Threading.Timer.Change(TimeSpan dueTime, TimeSpan period)
at MongoDB.Driver.Core.Clusters.Cluster.EnterServerSelectionWaitQueue()
at MongoDB.Driver.Core.Clusters.Cluster.SelectServerHelper.WaitingForDescriptionToChange()
at MongoDB.Driver.Core.Clusters.Cluster.SelectServer(IServerSelector selector, CancellationToken cancellationToken)
at MongoDB.Driver.MongoClient.AreSessionsSupportedAfterServerSelection(CancellationToken cancellationToken)
at MongoDB.Driver.MongoClient.AreSessionsSupported(CancellationToken cancellationToken)
at MongoDB.Driver.OperationExecutor.StartImplicitSession(CancellationToken cancellationToken)
at MongoDB.Driver.MongoCollectionImpl`1.UsingImplicitSession[TResult](Func`2 func, CancellationToken cancellationToken)
at MongoDB.Driver.MongoCollectionImpl`1.FindSync[TProjection](FilterDefinition`1 filter, FindOptions`2 options, CancellationToken cancellationToken)
at MongoDB.Driver.FindFluent`2.ToCursor(CancellationToken cancellationToken)
at MongoDB.Driver.IAsyncCursorSourceExtensions.ToList[TDocument](IAsyncCursorSource`1 source, CancellationToken cancellationToken)
at Eisengrind.FFst.Server.Core.Storage.BankOrders.MongoDB.Database..ctor(IFFDatabase database) in /builds/51st-state/server/src/server/Core/Storage/BankOrders/MongoDB/Database.cs:line 25
at lambda_method(Closure , Object[] )
at Autofac.Core.Activators.Reflection.ConstructorParameterBinding.Instantiate()
--- End of inner exception stack trace ---
at Autofac.Core.Activators.Reflection.ConstructorParameterBinding.Instantiate()
at Autofac.Core.Activators.Reflection.ReflectionActivator.ActivateInstance(IComponentContext context, IEnumerable`1 parameters)
at Autofac.Core.Resolving.InstanceLookup.CreateInstance(IEnumerable`1 parameters)
--- End of inner exception stack trace ---
at Autofac.Core.Resolving.InstanceLookup.CreateInstance(IEnumerable`1 parameters)
at Autofac.Core.Resolving.InstanceLookup.Execute()
at Autofac.Core.Resolving.ResolveOperation.GetOrCreateInstance(ISharingLifetimeScope currentOperationScope, ResolveRequest request)
at Autofac.Core.Resolving.ResolveOperation.Execute(ResolveRequest request)
at Autofac.ResolutionExtensions.TryResolveService(IComponentContext context, Service service, IEnumerable`1 parameters, Object& instance)
at Autofac.ResolutionExtensions.ResolveService(IComponentContext context, Service service, IEnumerable`1 parameters)
at Eisengrind.FFst.Server.Components.ComponentLoader.LoadComponents() in /builds/51st-state/server/src/server/Components/ComponentLoader.cs:line 25
at Eisengrind.FFst.Server.Bootstrapper.Bootstrapper.OnStart() in /builds/51st-state/server/src/server/Bootstrapper/Bootstrapper.cs:line 98
```

**Comment by DimaaIO** [comment]
The same problem occurs with redis.

```[20:12:46] alt:V Server, build #1160, branch release
[20:12:46] Starting LuckyV Server on 0.0.0.0:7788
[20:12:47] coreclr-module: version found: 3.1.1
[20:12:47] coreclr-module: greatest version: 3.1.1
[20:12:47] Loading resource gamemode
Threw a exception while executing constructor: + System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> StackExchange.Redis.RedisConnectionException: It was not possible to connect to the redis server(s).
   at StackExchange.Redis.ConnectionMultiplexer.ConnectImpl(Object configuration, TextWriter log) in C:\projects\stackexchange-redis\src\StackExchange.Redis\ConnectionMultiplexer.cs:line 1003
   at LuckyV.Server.Startup.ConfigureServices(IServiceCollection services) in /app/Server/LuckyV.Server/Startup.cs:line 48
   at LuckyV.Server.Lucky..ctor() in /app/Server/LuckyV.Server/Lucky.cs:line 36
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeConstructorInfo.Invoke(BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at AltV.Net.ResourceLoaders.AssemblyLoader.FindType[T](IEnumerable`1 assemblies, T& t)```

**Comment by architectlog** [comment]
I have the same error in an asynchronous function that checks a connection token. It was still working yesterday.

My server is running on a docker image that uses Microsoft's dotnet environment. cf: https://github.com/architectlog/altv-docker/blob/master/Dockerfile

**Comment by FabianTerhorst** [comment]
Fixed in latest altv dev build.

**Comment by Lhoerion** [comment]
~~Added~~/fixed/~~changed~~ in build #1167
